### PR TITLE
Add more haptic feedback varieties

### LINKED ISSUE #13710: Add more haptic feedback diversity
We currently have one type haptic feedback that triggers a medium alarm style haptic vibrate on iOS and a shorter knock on Android. 

Add more variety and try to map them between platforms 
https://developer.android.com/reference/android/view/HapticFeedbackConstants.html
https://developer.apple.com/documentation/uikit/animation_and_haptics

### PR DESCRIPTION
**Pull request by xster**
Fixes https://github.com/flutter/flutter/issues/13710

Needs to follow https://github.com/flutter/engine/pull/4699

### CODE REVIEW

**Comment by cbracken** [comment]
Tests please! :)

File: packages/flutter/test/services/haptic_feedback_test.dart [hunk 1]
```diff
@@ -18,4 +18,32 @@ void main() {
...
+    final RegExp functionName = new RegExp(r".*\'(\w+)\'");
+
+    for (int i = 0; i < hapticFunctions.length; i++) {
+      await Function.apply(hapticFunctions[i], null);
+      expect(log, hasLength(i + 1));
+      expect(
+        log.last,
+        isMethodCall(
+          'HapticFeedback.vibrate',
+          arguments: 'HapticFeedbackType.${functionName.firstMatch(hapticFunctions[i].toString()).group(1)}',
```
Comment (xster) [inline packages/flutter/test/services/haptic_feedback_test.dart]: totally soliciting less hacky ways of doing this

Comment (cbracken) [reply packages/flutter/test/services/haptic_feedback_test.dart]: Simpler and less clever might be write a utility function:
```dart
void expectHapticFeedbackCall(Function hapticFunction, String expectedType) {
     final List<MethodCall> log = <MethodCall>[];
 
     SystemChannels.platform.setMockMethodCallHandler((MethodCall methodCall) async {
       log.add(methodCall);
     });
 
     await hapticFunction();
 
     expect(log, hasLength(1));
     expect(log.single, isMethodCall('HapticFeedback.vibrate', arguments: expectedType));
}
```

Then just call that for each of the functions.

Comment (xster) [reply packages/flutter/test/services/haptic_feedback_test.dart]: done

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

**Pull request merged**